lib/github_classroom/lti/request_signing.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module GitHubClassroom
+  module LTI
+    module RequestSigning
+      def signed_auth_header(endpoint, consumer_key, secret)
+        consumer = OAuth::Consumer.new(consumer_key, secret, site: endpoint)
+        req = Net::HTTP::Get.new(endpoint, nil)
+
+        auth_signer = OAuth::Client::Helper.new(req, consumer: consumer, request_uri: endpoint)
+        auth_signer.hash_body
+
+        auth_signer.header
+      end
+
+      def signed_request(endpoint, consumer_key, secret, headers: {})
+        headers = { "Authorization": signed_auth_header(endpoint, consumer_key, secret) }.merge(headers)
+        opts = {
+          url: endpoint,
+          headers: headers
+        }
+
+        Faraday.new(opts) do |conn|
+          conn.response :raise_error
```

Comment:
Faraday has error handling for requests/response built in as a [middleware](https://lostisland.github.io/faraday/middleware/list), so we may as well use that for error handling rather than re-invent the wheel